spec/util/test.js

```diff
@@ -72,15 +73,25 @@ module.exports.beforeEach = function(testCase, env) {
     if (env) {
         env.ircMock._reset();
         env.clientMock._reset();
-        env.main = proxyquire("../../lib/main.js", {
-            "matrix-appservice": {
-                AppService: MockAppService,
-                "@global": true
-            },
-            "matrix-js-sdk": env.clientMock,
-            "irc": env.ircMock
-        });
-        env.mockAppService = MockAppService.instance();
+
+        let done = () => {
+            env.main = proxyquire("../../lib/main.js", {
+                "matrix-appservice": {
+                    AppService: MockAppService,
+                    "@global": true
+                },
+                "matrix-js-sdk": env.clientMock,
+                "irc": env.ircMock
+            });
+            env.mockAppService = MockAppService.instance();
+        };
+
+        if (env.main) {
+            env.main.killBridge().then(done);
```

Comment:
It seems to me that ```beforeEach``` should return a deferred promise, but this will mean changing all the tests to yield on it. Or I could put an optional callback in ```killBridge``` and give it ```done```?